docs/proposals-accepted/202206-block-series-deletion-compactor.md

```diff
@@ -0,0 +1,219 @@
+
+Overall, this is not efficient as rewrite is a different step from compaction so blocks might need to be compacted multiple times, causing [write amplification](https://en.wikipedia.org/wiki/Write_amplification) issues. We also need to add this step to the planner and make sure it works together with existing compaction.
+
+### Additional Design Considerations
+
+#### Store Gateway Masks Series based on Tombstones
+
+Like block metadata, the store gateway also fetches tombstones in a configurable interval, which means some delay is expected: user creates a tombstone but she is still able to query the data to be deleted.
+
+There are different ways to mask series based on tombstones. In the previous proposal, we didn't mention the detailed implementation. We will discuss two options here:
+
+##### Option 1: Mask Series At Query Time
+
+During query time, the store gateway decides what blocks to query based on the request min, max time and label matchers. Meanwhile, it can also decide what tombstones are going to be applied for the selected blocks based on tombstone time range and label matchers.
+
+Then during the query time, it checks the series label sets against the tombstone matchers, as well as the chunks and tombstone time range to mask the series data.
+
+Pros:
+* Easy to implement and very straightforward
+* Less memory usage.
+
+Cons:
+* Slower at query time
+
+##### Option 2: Mask Series At Tombstone Sync Time
+
+During the tombstone file sync, store gateway can also eagerly apply the tombstones to all blocks, calculate the matched postings and time range for each block. The result is the same as `MemTombstone` we mentioned before.
+
+```go
+// Result for the processed result for each tombstone, each block.
+type MemTombstones struct {
+	intvlGroups map[storage.SeriesRef]Intervals
+	mtx         sync.RWMutex
+}
+
+// Result for the processed result for each block.
+// Key: global tombstone ID, Value: processed result for each tombstone.
+type BlockTombstones map[ulid.ULID]*MemTombstones
+```
+
+Since all the blocks and tombstones are immutable, this result can be easily cached in memory or in filesystem. Ideally for each tombstone, each block we just need to calculate once. Because `BlockTombstones` is a map, it is easy to add or delete one key if users want to add more tombstones or remove one tombstone.
+
+During the query time, the cached tombstone will be used. As it knows which `SeriesRef`, which intervals to apply, it is very straightforward to mask series.
+
+Pros:
+* More efficient during query time, less processing required.
+* As data can be cached there is not much computation required.
+
+Cons:
+* More memory required.
+* If read is much less than write, it is inefficient as the pre-calculated result won't be used much.
+
+##### Considerations
```

Comment:
What this means? We already added `### Proposed Option"`.

Is this section still relevant?